.github/workflows/ci-cd.yml

```diff
@@ -241,6 +242,14 @@ jobs:
       uses: py-actions/py-dependency-install@v4
       with:
         path: requirements/test.txt
+    - name: Install Cython nightly on free-threading
```

Comment:
I assume this can also be removed once Cython 3.1 is out? Can we get a TODO comment here as well.

Reply:
Actually, will 3.1.0a1 work? Would probably simplify some of this, as that's already on PyPI.

Reply:
Wasn't this the one that broke yarl during the sprints?

Reply:
Yes